pkg/apis/pipeline/v1/taskrun_types.go

```diff
@@ -0,0 +1,473 @@
+	RetriesStatus []TaskRunStatus `json:"retriesStatus,omitempty"`
+
+	// TaskRunResults are the list of results written out by the task's containers
+	// +optional
+	// +listType=atomic
+	TaskRunResults []TaskRunResult `json:"taskResults,omitempty"`
+
+	// The list has one entry per sidecar in the manifest. Each entry is
+	// represents the imageid of the corresponding sidecar.
+	// +listType=atomic
+	Sidecars []SidecarState `json:"sidecars,omitempty"`
+
+	// TaskSpec contains the Spec from the dereferenced Task definition used to instantiate this TaskRun.
+	TaskSpec *TaskSpec `json:"taskSpec,omitempty"`
+}
+
+// TaskRunStepOverride is used to override the values of a Step in the corresponding Task.
+type TaskRunStepOverride struct {
+	// The name of the Step to override.
+	Name string `json:"name"`
+	// The resource requirements to apply to the Step.
+	Resources corev1.ResourceRequirements `json:"resources"`
+}
+
+// TaskRunSidecarOverride is used to override the values of a Sidecar in the corresponding Task.
+type TaskRunSidecarOverride struct {
+	// The name of the Sidecar to override.
+	Name string `json:"name"`
+	// The resource requirements to apply to the Sidecar.
+	Resources corev1.ResourceRequirements `json:"resources"`
+}
+
+// GetGroupVersionKind implements kmeta.OwnerRefable.
+func (*TaskRun) GetGroupVersionKind() schema.GroupVersionKind {
+	return SchemeGroupVersion.WithKind(pipeline.TaskRunControllerName)
+}
+
+// GetStatusCondition returns the task run status as a ConditionAccessor
+func (tr *TaskRun) GetStatusCondition() apis.ConditionAccessor {
+	return &tr.Status
+}
+
+// GetCondition returns the Condition matching the given type.
+func (trs *TaskRunStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return taskRunCondSet.Manage(trs).GetCondition(t)
+}
+
+// InitializeConditions will set all conditions in taskRunCondSet to unknown for the TaskRun
+// and set the started time to the current time
+func (trs *TaskRunStatus) InitializeConditions() {
+	started := false
+	if trs.StartTime.IsZero() {
+		trs.StartTime = &metav1.Time{Time: time.Now()}
+		started = true
+	}
+	conditionManager := taskRunCondSet.Manage(trs)
+	conditionManager.InitializeConditions()
+	// Ensure the started reason is set for the "Succeeded" condition
+	if started {
+		initialCondition := conditionManager.GetCondition(apis.ConditionSucceeded)
+		initialCondition.Reason = TaskRunReasonStarted.String()
+		conditionManager.SetCondition(*initialCondition)
+	}
+}
+
+// SetCondition sets the condition, unsetting previous conditions with the same
+// type as necessary.
+func (trs *TaskRunStatus) SetCondition(newCond *apis.Condition) {
+	if newCond != nil {
+		taskRunCondSet.Manage(trs).SetCondition(*newCond)
+	}
+}
+
+// StepState reports the results of running a step in a Task.
+type StepState struct {
+	corev1.ContainerState `json:",inline"`
+	Name                  string `json:"name,omitempty"`
+	ContainerName         string `json:"container,omitempty"`
```

Comment:
I think for `StepState` and `SidecarState` we can rename ContainerName -> Container to fix the API rule violation.